courses/machine_learning/deepdive2/structured/solutions/1_explore_babyweight.ipynb

```diff
@@ -0,0 +1,536 @@
+{
```

Comment:
<p>This is a security risk and there should either be a warning to students, or should be done in a way that prevents SQL injection</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/665/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>These are learning how to do ML notebooks, we're not trying to protect against every form of SQL injection. This is just trying to get a simple concept across, it is up to the user to harden any code they want to use in the real world outside of the classroom.</p>

Reply:
Then do the former and add a quick note and [a link](https://cloud.google.com/bigquery/docs/parameterized-queries). I've had multiple students flag to me that this is not industry best practice. Let's put them at ease that we know what we're doing :)

Reply:
You can't parameterize column names it seems, those are not scalars, arrays of scalars, or structs of scalars. Therefore, I can't do the former to leave a note or link because the link's method doesn't work for column names. And I can't do the latter because it doesn't work.

From the documentation:
Parameters cannot be used as substitutes for identifiers, column names, table names, or other parts of the query.

Reply:
Right, Bigquery doesn't have an easy solution, but that doesn't make this code safe.
What you could do is make a set of acceptable column names, and see if the passed variable is in the set, and mention this is a crude way of preventing SQL injection for instructional purposes. It should be, at most, three lines of code to be added.

Reply:
<p>Added SQL Injection protection to code.</p>